cpp/ql/test/library-tests/ir/ir/try_except.c

```diff
@@ -0,0 +1,28 @@
+// semmle-extractor-options: --microsoft
+
+void ProbeFunction();
+void sink();
+
+void f() {
+  int x, y = 0;
+  __try {
+    ProbeFunction(0);
+    x = y;
+    ProbeFunction(0);
+  }
+  __except (0) {
+    sink(x);
+  }
+}
+
+void g() {
+  int x, y = 0;
+  __try {
+    ProbeFunction(0);
+    x = y;
+    ProbeFunction(0);
+  }
+  __finally {
+    sink(x);
+  }
+}
```

Comment:
Looking at syntax zoo, could we also add something like the following here:
```cpp
void h(int b) {
    __try {
        if (b) {
            throw 1;
        }
    }
    __except (1) {
        sink(x)
    }
}
```

Reply:
Good call. I've added it in 5fa968138c73eafc4017c9de35c8029172b55c12.

Reply:
Ahh, but with `AfxThrowMemoryException` it degenerates to the first case above. Could we add a C++ file with an actual `throw`?

Reply:
Done in a974cb186168d13af5da3fd29b092167fa510ffe. It looks like the test behaves as expected: we generate a `ThrowValue` instruction that correctly transfers flow to the handler block.

Reply:
> we generate a `ThrowValue` instruction that correctly transfers flow to the handler block.

Is that actually what should be happening? I really don't know...

Reply:
https://learn.microsoft.com/en-us/cpp/cpp/structured-exception-handling-c-cpp?view=msvc-170#example

Reply:
So, this indeed seems to be the correct behaviour.